tool/tsh/common/kube.go

```diff
@@ -979,51 +979,74 @@ func (c *kubeLSCommand) run(cf *CLIConf) error {
 	}
 
 	selectedCluster := selectedKubeCluster(currentTeleportCluster, getKubeConfigPath(cf, ""))
+	err = c.showKubeClusters(cf.Stdout(), kubeClusters, selectedCluster)
+	return trace.Wrap(err)
+}
+
+func (c *kubeLSCommand) showKubeClusters(w io.Writer, kubeClusters types.KubeClusters, selectedCluster string) error {
 	format := strings.ToLower(c.format)
 	switch format {
 	case teleport.Text, "":
-		var (
-			t       asciitable.Table
-			columns = []string{"Kube Cluster Name", "Labels", "Selected"}
-			rows    [][]string
-		)
-
-		for _, cluster := range kubeClusters {
-			var selectedMark string
-			if cluster.GetName() == selectedCluster {
-				selectedMark = "*"
-			}
-			rows = append(rows, []string{
-				cluster.GetName(),
-				common.FormatLabels(cluster.GetAllLabels(), c.verbose),
-				selectedMark,
-			})
-		}
-
-		if c.quiet {
-			t = asciitable.MakeHeadlessTable(2)
-			for _, row := range rows {
-				t.AddRow(row[:2])
-			}
-		} else if c.verbose {
-			t = asciitable.MakeTable(columns, rows...)
-		} else {
-			t = asciitable.MakeTableWithTruncatedColumn(columns, rows, "Labels")
-		}
-		fmt.Fprintln(cf.Stdout(), t.AsBuffer().String())
+		showKubeClustersAsText(w, kubeClusters, selectedCluster, c.quiet, c.verbose)
```

Comment:
we should keep the same logic as below
```suggestion
		out, err:=serializeKubeClustersAsText(kubeClusters, selectedCluster, c.quiet, c.verbose)
		if err != nil {
			return trace.Wrap(err)
		}
		fmt.Fprintln(w, out)
```

Reply:
[05ae6df](https://github.com/gravitational/teleport/pull/30149/commits/05ae6df632f2126e2bf7f8a36dc6dd8482262dbf)